k2/csrc/tensor_ops_test.cu

```diff
@@ -325,10 +326,69 @@ static void TestSimpleRaggedIndexSelect1D() {
   }
 }
 
+
 TEST(Index, SimpleRaggedIndexSelect1D) {
   TestSimpleRaggedIndexSelect1D<float>();
   TestSimpleRaggedIndexSelect1D<double>();
   TestSimpleRaggedIndexSelect1D<int32_t>();
 }
 
+
+TEST(Tensor, DiscountedCumSum) {
+  for (int32_t i = 0; i < 4; i++) {
+    int32_t M = RandInt(0, 1000),
+        T = RandInt(0, 2000);  // TODO: increase.
+    while (M * T > 10000) {  // don't want test to take too long.
+      M /= 2;
+      T /= 2;
```

Comment:
It may happen that outside the while loop,  one of `M` or `T` is 0, which will
make the test fail.

Reply:
In general things are supposed to work for zero-size arrays...